docs/usage_guide/benchmarking.rst

```diff
@@ -0,0 +1,180 @@
+Benchmarking Modin
+==================
+
+Summary
+-------
+To benchmark a single Modin function, often turning on the
+:doc:`configuration variable </flow/modin/config>` variable
+:code:`BenchmarkMode` will suffice.
+
+There is no simple way to benchmark more complex Modin workflows, though
+benchmark mode or calling ``repr`` on Modin objects may be useful. The
+:doc:`Modin logs </usage_guide/advanced_usage/modin_logging>` may help you
+identify bottlenecks in your code, and they may also help profile the execution
+of each Modin function.
+
+Modin's execution and benchmark mode
+------------------------------------
+
+Most of Modin's execution happens asynchronously, i.e. in separate processes that run
+independently of the main program flow. Some execution is also lazy, meaning that it
+doesn't start immediately once the user calls a Modin function. While Modin provides
+the same API as pandas, lazy and asynchronous execution can often make it hard to
+tell how much time each Modin function call takes, as well as to compare Modin's
+performance to pandas and other similar libraries.
+
+.. note::
+    All examples in this doc use the system specified at the bottom of this page.
+
+Consider the following ipython script:
+
+.. code-block:: python
+
+    import modin.pandas as pd
+    from modin.config import MinPartitionSize
+    import time
+    import ray
+
+    ray.init()
+    df = pd.DataFrame(list(range(MinPartitionSize.get())))
+    %time result = df.applymap(lambda x: time.sleep(0.2) or x)
+    %time print(result)
+
+
+Modin takes just 2.68 milliseconds for the ``applymap``, and 6.78 seconds to print
+the result. However, if we run this script in pandas by replacing
+:code:`import modin.pandas as pd` with :code:`import pandas as pd`, the ``applymap``
+takes 6.5 seconds, and printing the result takes just 3.36 milliseconds.
+
+Both pandas and Modin start executing the ``applymap`` as soon as the interpreter
+evalutes it. While pandas blocks until the ``applymap`` has finished, Modin just kicks
+off asynchronous functions in remote ray processes. Printing the function result
+is fairly fast in pandas and Modin, but before Modin can print the data, it has to
+wait until all the remote functions complete.
+
+To time how long Modin takes for a single operation, you should typically use
+benchmark mode. You can turn on benchmark mode on at any point as follows:
+
+.. code-block:: python
+
+    from modin.config import BenchmarkMode
+    BenchmarkMode.put(True)
+
+Rerunning the script above with benchmark mode on, the Modin ``applymap`` takes
+6.79 seconds, and the ``print`` takes 5.46 milliseconds. These timings better
+reflect where Modin is spending its execution time.
```

Comment:
Is it worth mentioning here the reason why Modin takes a similar amount of time as pandas (I'm assuming it's small dataframe size), instead of being faster?